core/server/data/migrations/versions/4.9/01-add-default-product-portal-products.js

```diff
@@ -0,0 +1,56 @@
+const {createTransactionalMigration} = require('../../utils');
+const logging = require('../../../../../shared/logging');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const products = await knex
+            .select('id')
+            .from('products');
+
+        if (products.length === 0) {
+            logging.warn(`Skipping adding default product to portal_products, no default product exists`);
+            return;
+        }
+
+        const defaultProduct = products[0];
+
+        const portalProductSetting = await knex('settings')
+            .where('key', 'portal_products')
+            .select('value')
+            .first();
+
+        if (!portalProductSetting) {
+            logging.info('Skipping adding default product to portal_products, setting does not exist');
+            return;
+        }
+
+        if (portalProductSetting.length > 0) {
```

Comment:
I think you need to `JSON.parse` the value first, otherwise you're checking the length of a string!

Reply:
Good catch, updated! ✅